```
/home/coder/cccl/lib/cmake/libcudacxx/../../../libcudacxx/include/cuda/std/__pstl/cuda/unique.h:83:55: error: ‘template<class InputIteratorT, class FlagsInputIteratorT, class SelectedOutputIteratorT, class NumSelectedIteratorT, class SelectOpT, class EqualityOpT, class OffsetT, cub::_V_300500_SM_750_800_900::SelectImpl SelectionOpt, class PolicyHub> struct cub::_V_300500_SM_750_800_900::DispatchSelectIf’ is deprecated [-Werror=deprecated-declarations]
       using DispatchUnique = CUB_NS_QUALIFIER::DispatchSelectIf<
                                                         ^~~~~~~~        
  /home/coder/cccl/lib/cmake/cub/../../../cub/cub/device/dispatch/dispatch_select_if.cuh:457:80: note: declared here
   struct CCCL_DEPRECATED_BECAUSE("Please use DeviceSelect or DevicePartition") DispatchSelectIf
                                                                                  ^~~~~~~~~~~~~~  
```
@miscco that's on you :) Do you need to use the Dispatch struct because of temporary storage query again?

--- a/cub/cub/device/device_partition.cuh
+++ b/cub/cub/device/device_partition.cuh
@@ -115,9 +115,10 @@ struct DevicePartition
     [[nodiscard]] _CCCL_HOST_DEVICE_API constexpr auto operator()(::cuda::compute_capability cc) const -> SelectPolicy
     {
       // the user-provided policy selector returns a PartitionPolicy, the default one a SelectPolicy
+      using policy_t = ::cuda::std::remove_cvref_t<decltype(PolicySelector{}(cc))>;
+      static_assert(
+        ::cuda::std::is_same_v<policy_t, PartitionPolicy> || ::cuda::std::is_same_v<policy_t, SelectPolicy>);
       const auto policy = PolicySelector{}(cc);
-      static_assert(::cuda::std::is_same_v<decltype(policy), PartitionPolicy>
-                    || ::cuda::std::is_same_v<decltype(policy), SelectPolicy>);
       return SelectPolicy{
         policy.threads_per_block,
         policy.items_per_thread,

--- a/cub/test/CMakeLists.txt
+++ b/cub/test/CMakeLists.txt
@@ -184,6 +184,7 @@ function(
       "${test_src}" MATCHES "test_device_segmented_scan_api\\.cu$"
       OR "${test_src}" MATCHES "test_device_find_env_api\\.cu$"
       OR "${test_src}" MATCHES "test_device_partition_env_api\\.cu$"
+      OR "${test_src}" MATCHES "test_device_select_env_api\\.cu$"
     )
       if (
         "Clang" STREQUAL "${CMAKE_CXX_COMPILER_ID}"

--- a/cub/test/CMakeLists.txt
+++ b/cub/test/CMakeLists.txt
@@ -183,6 +183,7 @@ function(
     if (
       "${test_src}" MATCHES "test_device_segmented_scan_api\\.cu$"
       OR "${test_src}" MATCHES "test_device_find_env_api\\.cu$"
+      OR "${test_src}" MATCHES "test_device_partition_env_api\\.cu$"
     )
       if (
         "Clang" STREQUAL "${CMAKE_CXX_COMPILER_ID}"

